tests/hdfs_kerberos_test.go

```diff
@@ -0,0 +1,150 @@
+package tests
+
+import (
+	"fmt"
+	"github.com/mesosphere/kudo-spark-operator/tests/utils"
+	"github.com/stretchr/testify/suite"
+	"testing"
+	"time"
+)
+
+var (
+	resourceFolder         = "resources/hdfs-kerberos"
+	namespace              = "hdfs-kerberos"
+	hdfsKerberosDeployment = []Resource{
+		{
+			name: "configmaps",
+			path: "configmaps",
+		},
+		{
+			name: "volumes",
+			path: "volumes",
+		},
+		{
+			name: "kerberos",
+			path: "kerberos-deployment.yaml",
+			wait: true,
+		},
+		{
+			name: "hdfs-namenode",
+			path: "namenode-deployment.yaml",
+			wait: true,
+		},
+		{
+			name: "hdfs-datanode",
+			path: "datanode-deployment.yaml",
+			wait: true,
+		},
+	}
+	eventLogDir       = "hdfs://namenode.hdfs-kerberos.svc.cluster.local:9000/history"
+	hadoopTokenSecret = "hadoop-token"
+)
+
+type Resource struct {
+	name string
+	path string
+	wait bool
+}
+
+type HdfsIntegrationSuite struct {
+	operator utils.SparkOperatorInstallation
+	suite.Suite
+}
+
+func TestHdfsIntegrationSuite(t *testing.T) {
+	suite.Run(t, new(HdfsIntegrationSuite))
+}
+
+func (suite *HdfsIntegrationSuite) SetupSuite() {
+	utils.Kubectl("create", "ns", namespace)
+	// deploy KDC and HDFS
+	for _, resource := range hdfsKerberosDeployment {
+		utils.Kubectl("apply", "-f", fmt.Sprint(resourceFolder, "/", resource.path), "-n", namespace)
+
+		if resource.wait {
+			utils.Kubectl("wait", fmt.Sprint("deployments/", resource.name),
+				"--for=condition=available", "--timeout=3m", "-n", namespace)
+		}
+	}
+	nameNodePod, err := utils.Kubectl("get", "pods", "--selector=name=hdfs-namenode",
+		"--output=jsonpath={.items[*].metadata.name}", "-n", namespace)
+	if err != nil {
+		suite.FailNow("Error while getting a name of Namenode pod", err)
+	}
+	// run init script to copy test data to HDFS and export delegation token
+	utils.Kubectl("exec", nameNodePod, "-n", namespace, "--", "init.sh")
+
+	// copy delegation token from the pod to a local filesystem
+	utils.Kubectl("cp", fmt.Sprint(nameNodePod, ":", "tmp/hadoop.token"), "/tmp/hadoop.token", "-n", namespace)
+	utils.Kubectl("create", "secret", "generic", hadoopTokenSecret, "--from-file", "/tmp/hadoop.token", "-n", namespace)
```

Comment:
Looking through the changes in SHS deployment template and it looks like that Operator assumes that the file name in the secret is`hadoop.token` - we should reflect it in the docs.

Reply:
added explicit comment to the test code about `hadoop.token` secret key